site/src/pages/blocks/index.page.tsx

```diff
@@ -0,0 +1,105 @@
+import { PublishBlockSuccess } from "../../components/pages/blocks/publish-block-success";
+import { PageContainer } from "../../components/pages/dashboard/page-container";
+import { TopNavigationTabs } from "../../components/pages/dashboard/top-navigation-tabs";
+import { ListViewCard } from "../../components/pages/user/list-view-card";
+import { apiClient } from "../../lib/api-client";
+import { ExpandedBlockMetadata } from "../../lib/blocks";
+import { shouldAllowBlockPublishing } from "../../lib/config";
+import { formatUpdatedAt } from "../../util/html-utils";
+
+const BlocksPage: AuthWallPageContent = ({ user }) => {
+  const router = useRouter();
+  const [blocks, setBlocks] = useState<ExpandedBlockMetadata[]>([]);
+
+  useEffect(() => {
+    if (user.shortname) {
+      /** @todo replace this manual fetching with `SWR` or `React Query` for better DX/UX */
+      void apiClient
+        .getUserBlocks({
+          shortname: user.shortname.replace("@", ""),
+        })
+        .then((res) => setBlocks(res.data?.blocks || []));
+    }
+  }, [user]);
+
+  const recentlyCreatedBlock = useMemo(
+    () => blocks.find((block) => block.name === router.query.createdBlock),
+    [blocks, router],
+  );
+
+  const hasBlocks = !!blocks.length;
+
+  return (
+    <>
+      <Head>
+        <title>Block Protocol – Published Blocks</title>
+      </Head>
+
+      <TopNavigationTabs />
+
+      <PageContainer>
+        <Box display="flex" alignItems="center" gap={2} flexWrap="wrap" mb={4}>
+          <Typography variant="bpHeading2">Published Blocks</Typography>
+          {shouldAllowBlockPublishing && (
+            <LinkButton
+              startIcon={<FontAwesomeIcon icon={faPlus} />}
+              size="small"
+              squared
+              variant="tertiary"
+              color="gray"
+              href="/blocks/publish"
+            >
+              Add new block
+            </LinkButton>
+          )}
+        </Box>
+
+        {!!recentlyCreatedBlock && (
+          <PublishBlockSuccess
+            blockHref={`@${user.shortname}/blocks/${recentlyCreatedBlock?.name}`}
+          />
+        )}
+
+        <BlockListContainer hasBlocks={hasBlocks}>
+          {hasBlocks ? (
+            blocks.map((block) => (
+              <ListViewCard
+                key={block.name}
+                url={block.blockSitePath}
+                icon={block.icon}
+                title={block.displayName!}
+                description={block.description}
+                extraContent={
+                  <Box display="flex" gap={1.5}>
+                    <Link href={`/@${block.author}`}>@{block.author}</Link>
+                    <span>{`V${block.version}`}</span>
+                    <span>{formatUpdatedAt(block.lastUpdated)}</span>
+                  </Box>
+                }
+              />
+            ))
+          ) : (
+            <BlockListEmptyState />
```

Comment:
This will always flash on first load – can we introduce a loading state instead to avoid displaying "You haven't published any blocks" to everyone, even if they have?

Reply:
Thanks for the heads up, forgot about loading state somehow 🤦 
Addressed here https://github.com/blockprotocol/blockprotocol/pull/526/commits/f9968db0cde28c0704902c5535696e9e432e90b7